### DISCUSSION TITLE
How to handle useOptimistic with Framer Motion?

### DISCUSSION THREAD

**Discussion by bdrtsky** [motiondivision/motion#2768] [Q&A]
I know, v12 is still a WIP, but playing with new React, things kinda working, but have issues with Optimistic UI strategy. 

Let's say a simple TODO list. If we create item optimistically, it means that item will be rendering twice (once instantly optimistic, and later after server response). So using Framer Motion the way we used to seems not working anymore correctly, as animation will be triggered multiple times.

Wonder if anyone played with it already? @mattgperry did you tried it?

**Comment by bdrtsky** [comment]
I'll post sandbox soon if anyone interested to take a look 🙌

**Comment by aleksa-codes** [comment]
facing similar problem, but when trying to implement Optimistic updates with `@tanstack/react-query` and animations with `motion/react`. This post is the only thing I found related to it.

**Reply by paulschuetz** [reply]
haha funny! I am facing the same problem for a chat app :D
